Tracker/Storage/StorageCoreData.swift

```diff
@@ -0,0 +1,265 @@
+        for categoryCD in categoriesFromStorage {
+            categories.append(TrackerCategory(
+                title: categoryCD.title ?? "",
+                trackers: loadTrackers(category: categoryCD)
+            ))
+        }
+        return categories
+    }
+    
+    func loadTrackers(category: TrackerCategoryCD) -> [Tracker] {
+        var trackers: [Tracker] = []
+        var trackersFromStorage: [TrackerCD] = []
+        
+        trackerRequest.predicate = NSPredicate(format: "category.title == %@", category.title!)
+        do {
+            trackerRequest.sortDescriptors = [sortByTitle]
+            trackersFromStorage = try context.fetch(trackerRequest)
+        }
+        catch {
+            assertionFailure("Couln't load trackers from CoreData!")
+        }
+        for trackerCD in trackersFromStorage {
+            trackers.append(Tracker(
+                id: Int((trackerCD.trackerID)),
+                title: trackerCD.title ?? "",
+                schedule: trackerCD.schedule ?? "",
+                emoji: trackerCD.emoji ?? "",
+                color: Int(trackerCD.color)
+            ))
+        }
+        trackerRequest.predicate = nil
+        return trackers
+    }
+    
+// MARK: - Save categories & trackers
+    func saveCategories(categories: [TrackerCategory]) {
+        for category in categories {
+            categoryRequest.predicate = NSPredicate(format: "title == %@", category.title)
+            let searchCategory = try? context.fetch(categoryRequest).first
+            
+            if searchCategory == nil {
+                let newCategory = TrackerCategoryCD(context: context)
+                newCategory.title = category.title
+                for tracker in category.trackers {
+                    saveTracker(tracker: tracker, category: newCategory)
+                }
+            } else {
+                for tracker in category.trackers {
+                    saveTracker(tracker: tracker, category: searchCategory)
+                }
+            }
+        }
+        categoryRequest.predicate = nil
+        do {
+            try context.save()
+        }
+        catch {
+            assertionFailure("Couln't save categories to CoreData!")
+        }
+    }
+    
+    
+    func saveTracker(tracker: Tracker, category: TrackerCategoryCD?) {
+        trackerRequest.predicate = NSPredicate(format:"title == %@", tracker.title)
+        let searchTracker = try! context.fetch(trackerRequest)
+        if searchTracker.isEmpty {
+            let newTracker = TrackerCD(context: context)
+            newTracker.trackerID = Int64(trackerID())
+            newTracker.category = category
+            newTracker.title = tracker.title
+            newTracker.schedule = tracker.schedule
+            newTracker.emoji = tracker.emoji
+            newTracker.color = Int64(tracker.color)
+        } else {
+            let updatedTracker = searchTracker.first(where: { $0.category?.title == category?.title })
```

Comment:
будет чуть попороще выглядеть код, если изначально разыменовать опциональное значение
```suggestion
        } else if let updatedTracker = searchTracker.first(where: { $0.category?.title == category?.title })
```